cosmic_ray/mutating.py

```diff
@@ -49,26 +49,30 @@ def activation_record(self):
         """
         return self._activation_record
 
-    def visit_mutation_site(self, node, op):
+    def visit_mutation_site(self, node, op, count):
         """Potentially mutate `node`, returning the mutated version.
 
         `Operator` calls this when AST iteration reaches a
         potential mutation site. If that site is scheduled for
         mutation, the subclass instance will be asked to perform the
         mutation.
         """
-        if self._count == self._target:
-            self._activation_record = {
-                'operator': _full_module_name(op),
-                'occurrence': self._target,
-                'line_number': cosmic_ray.util.get_line_number(node)
-            }
-
-            old_node = node
-            node = op.mutate(old_node)
-            ast.copy_location(node, old_node)
-
-        self._count += 1
+        for idx in range(count):
```

Comment:
I'm not very confident in making this change b/c I don't understand this piece of code quite well, hence the above implementation. What puzzles me is how exactly we get to the point of calling this method ? As far as I understand it `self._target` is the N-th occurrence of a particular operator we want to mutate. After the changes the meaning of N is no longer the N-th occurrence but the N-th possible mutation for this operator. Where N ranges from 0 to num_operator * num_variations (you get what I mean). 

Even if I do the change you request above I will need the `idx` variable to pass to `mutate()` so it knows which particular variation to produce. I still need to loop over the range to get `idx`.